featurebyte/worker/task_executor.py

```diff
@@ -61,13 +69,20 @@ def run_async(coro: Awaitable[Any]) -> Any:
         thread = Thread(target=start_background_loop, args=(loop,), daemon=True)
         thread.start()
 
-    tasks = asyncio.all_tasks()
-    logger.debug("Asyncio tasks", extra={"num_tasks": len(tasks)})
+    logger.debug("Asyncio tasks", extra={"num_tasks": len(asyncio.all_tasks(loop=loop))})
+
+    logger.info("Start task", extra={"timeout": timeout})
     future = asyncio.run_coroutine_threadsafe(coro, loop)
-    event = gevent.event.Event()
-    future.add_done_callback(lambda _: event.set())
-    event.wait()
-    return future.result()
+    try:
+        with gevent.Timeout(seconds=timeout, exception=TimeoutError):
+            event = gevent.event.Event()
+            future.add_done_callback(lambda _: event.set())
+            event.wait()
+            return future.result()
```

Comment:
Seems like we could pass the `timeout` parameter to `future.result()` here (https://docs.python.org/3/library/asyncio-task.html#scheduling-from-other-threads).

Reply:
Tried that - it raises timeout error but unfortunately the async task continues to run in the asyncio thread